v8engine/src/main/java/cm/aptoide/pt/v8engine/analytics/Analytics.java

```diff
@@ -207,32 +213,73 @@ public void sendSpotAndShareEvents(String eventName, Map<String, String> attribu
+      private static final String USER_AGENT = "user_agent";
       static AppEventsLogger facebookLogger;
+      private static String utmSource = UNKNOWN;
+      private static String utmMedium = UNKNOWN;
+      private static String utmCampaign = UNKNOWN;
+      private static String utmContent = UNKNOWN;
+      private static String entryPoint = UNKNOWN;
 
-      public static void onCreate(android.app.Application application) {
+      public static void onCreate(android.app.Application application,
+          Converter.Factory converterFactory, OkHttpClient okHttpClient,
+          BodyInterceptor bodyInterceptor, SharedPreferences sharedPreferences,
+          TokenInvalidator tokenInvalidator) {
 
         //Integrate FacebookSDK
         FacebookSdk.sdkInitialize(application);
         AppEventsLogger.activateApp(application);
         facebookLogger = AppEventsLogger.newLogger(application);
+        FirstLaunchAnalytics firstLaunchAnalytics =
+            new FirstLaunchAnalytics(facebookLogger, Analytics.getInstance());
         Observable.fromCallable(() -> {
           AppEventsLogger.setUserID(((V8Engine) application).getIdsRepository()
               .getUniqueIdentifier());
           return null;
         })
-            .filter(__ -> SecurePreferences.isFirstRun(
-                SecurePreferencesImplementation.getInstance(application.getApplicationContext(),
-                    PreferenceManager.getDefaultSharedPreferences(application))))
-            .doOnNext(__ -> setupDimensions(application))
+            .filter(firstRun -> SecurePreferences.isFirstRun(sharedPreferences))
+            .doOnNext(dimensions -> setupDimensions(application))
+            .doOnNext(facebookFirstLaunch -> firstLaunchAnalytics.sendFirstLaunchEvent(utmSource,
+                utmMedium, utmCampaign, utmContent, entryPoint))
+            .flatMap(facebookFirstLaunch -> {
+              UTMTrackingBuilder utmTrackingBuilder =
+                  new UTMTrackingBuilder(getTracking(application), getUTM());
+              BIUTMAnalyticsRequestBody body =
+                  new BIUTMAnalyticsRequestBody(utmTrackingBuilder.getUTMTrackingData());
+              return BIUTMAnalyticsRequest.of(BI_ACTION, EVENT_NAME, CONTEXT, body, bodyInterceptor,
+                  okHttpClient, converterFactory, sharedPreferences, tokenInvalidator)
+                  .observe()
+                  .observeOn(Schedulers.io());
+            })
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe(o -> {
-            }, Throwable::printStackTrace);
+            }, throwable -> Logger.e(TAG, throwable.getMessage()));
+      }
+
+      private static UTM getUTM() {
+        return new UTM(utmSource, utmMedium, utmCampaign, utmContent, entryPoint);
       }
 
       private static void setupDimensions(android.app.Application application) {
         if (!checkForUTMFileInMetaINF(application)) {
```

Comment:
this logic should be separated? One method to update utms, other to check if it has utms